### PR TITLE
feat: support :nth-child odd/even pseudo-class

### PR DESCRIPTION
**Pull request by redonkulus**
Requested via https://github.com/orgs/acss-io/discussions/525

Adds pseudo-class support for `:nth-child(odd)` and `:nth-child(even)`. Note: Functional syntax support was not added for now, as the additional `()` maybe cause parser problems.

Screenshot of how it would be used:
Screen Shot 2022-10-28 at 8 24 25 AM

### CODE REVIEW

File: docs/guides/syntax.md [hunk 1]
```diff
@@ -151,13 +151,43 @@ Optional.
...
+-   `:placeholder-shown` => `:ps`,
+-   `:read-only` => `:ro`,
+-   `:read-write` => `:rw`,
+-   `:required` => `:req`,
+-   `:right` => `:r`,
+-   `:root` => `:rt`,
+-   `:scope` => `:s`,
+-   `:target` => `:t`,
+-   `:valid` => `:va`,
+-   `:visited` => `:vi`,
```
Comment (snyamathi) [inline docs/guides/syntax.md:190]: Would this be easier digested as a table

**Review by snyamathi — DISMISSED** [review]

File: docs/guides/syntax.md (hunk 1 above)
Comment (redonkulus) [reply docs/guides/syntax.md:190]: Hmm perhaps, I can try moving it to that

Comment (redonkulus) [reply docs/guides/syntax.md]: fixed https://github.com/acss-io/atomizer/pull/527/commits/3c5b556fe169d4f946dc290ccdee13964ea82300

**Review by snyamathi — APPROVED** [review]

**Comment by carriemorrison** [comment]
Do we think having only `odd` or `even` is useful? very limited use cases.. like styling table rows lol

**Comment by redonkulus** [comment]
The :nth-child fractional support would require atomizer parser updates to support the paranthesis in the class name. We haven't done that before for any other pseudo class or element.

**Comment by carriemorrison** [comment]
I guess these type of atomic rules are only useful for v simple loops since your going to add the SAME rule to each child element, like in your example.

```
 <ul>
    <li class="Bgc(red):odd">First item</li>
    <li class="Bgc(red):odd">Second item</li>
    <li class="Bgc(red):odd">Third item</li>
</ul>
```

Otherwise you could just alter the markup

```
            <ul>
                <li class="Bgc(red)">First item</li>
                <li class="">Second item</li>
                <li class="Bgc(red)">Third item</li>
            </ul>
```

I guess you could output both without any logic whatsoever

```
<ul>
    <li class="Bgc(red):odd Bgc(blue):even">First item</li>
    <li class="Bgc(red):odd Bgc(blue):even">Second item</li>
    <li class="Bgc(red):odd Bgc(blue):even">Third item</li>
</ul>
```

I personally would just use css for advanced logic, if we feel  having `odd` and `even` only is helpful then sure we can add this :)

**Comment by redonkulus** [comment]
Ya I agree odd/even are not really that beneficial when outputting on every child. I think having support for `:nth-child()` would be better. However, that will take some time to update the syntax and test is thoroughly. I'm going to close this PR for now.

**Pull request closed without merging**